proto/umee/leverage/v1/tx.proto

```diff
@@ -266,3 +271,30 @@ message MsgGovUpdateRegistry {
 
 // MsgGovUpdateRegistryResponse defines the Msg/GovUpdateRegistry response type.
 message MsgGovUpdateRegistryResponse {}
+
+// MsgGovUpdateSpecialAssetPairs defines the Msg/GovUpdateSpecialAssetPairs request type.
+message MsgGovUpdateSpecialAssetPairs {
+  option (gogoproto.equal)            = true;
+  option (gogoproto.goproto_getters)  = false;
+  option (gogoproto.goproto_stringer) = false;
+  option (cosmos.msg.v1.signer)       = "authority";
+
+  // authority is the address of the governance account.
+  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  // sets are bidirectional groups of special asset pairs. Creating a special asset
+  // set causes all assets in the set to have a certain collateral weight when borrowing
+  // against each other (but not looping with themselves). Overrides any existing
+  // special asset pairs between assets in the set. Using a collateral weight
+  // of zero will clear all existing special pairs in the set instead.
+  repeated SpecialAssetSet sets = 2 [(gogoproto.nullable) = false];
+
+  // pairs are new or updated special asset pairs. Updating a special asset pair's
+  // collateral weight to zero deletes the pair instead.
+  // These pairs will be applied after any sets above when passing a proposal,
+  // so they can be used for more granular control.
+  repeated SpecialAssetPair pairs = 3 [(gogoproto.nullable) = false];
```

Comment:
Using only sets has disadvantages:
- Forces us to allow / disallow looping for all assets involved in any sets at once, and requires a chain upgrade to fix
- Doesn't allow fine grained or directional control of assets (which we should keep just in case)
- Uniqueness problems (overlapping sets with different parameters)
- Iteration problems (no prefix tricks to select only assets relevant to a position)

But I agree that for creating a pair in the most general sense, we would use a 2-asset set.